Learnosity/learnosity-sdk-php
# [BUILD] fix docker configuration for the Quickstart application

### Summary:
This PR addresses issues with the PHP SDK running within a Docker container for the Quickstart application, specifically related to domain whitelisting and API access. The following changes were made to resolve these issues:

### Issues Addressed:

1. Whitelisting Failure:
- The PHP development server was previously set to bind to 0.0.0.0 to allow external connections. Since 0.0.0.0 is not a valid IP address, the Learnosity API calls failed as it was not whitelisted.

2. Mismatch Between Server Address and Requesting IP:
- After switching to 127.0.0.1 for local binding, the server configuration still led to a mismatch between the server’s address and the requesting IP address when running the command php -S $(LOCALHOST):8000. This mismatch caused signature security checks to fail.

### Proposed Solution:
To resolve the issues:

- A new Docker container configuration has been implemented with Nginx and PHP, where the server name is explicitly set to localhost. This ensures:
  - The Quickstart application’s API requests pass through the domain whitelist check.
  - The server and requesting IP addresses match, allowing the security signature checks to pass.

Additionally, customers can still run the Quickstart application locally without Docker. This can be tested by setting LRN_SDK_NO_DOCKER=false, allowing the PHP server to run independently of Docker.

### Acceptance Criteria:

- The Docker container with both Nginx and PHP containers correctly binds to localhost, the Quickstart application passes both whitelisting and security signature checks.
- Customers should be able to run make quickstart locally, without Docker, and successfully start the PHP server serving the Quickstart application.

### QA Testing:

1. Run make quickstart with Docker installed.
   - Verify Docker containers start correctly and serve the Quickstart app without errors. 
   - Ensure whitelisting and security checks pass.

2. Run make quickstart without Docker (using a local PHP server).
    - Ensure PHP and Composer are installed locally.
    - Run the command make LRN_SDK_NO_DOCKER=false quickstart and verify that the PHP server starts and serves the Quickstart app without errors.


## Checklist



- [x] All testsuite passes
- [x] `make dist` completed successfully
